x-pack/elastic-agent/pkg/agent/transpiler/vars.go

```diff
@@ -15,7 +15,7 @@ import (
 	"github.com/elastic/beats/v7/libbeat/common"
 )
 
-var varsRegex = regexp.MustCompile(`\${([\p{L}\d\s\\\-_|.'"]*)}`)
+var varsRegex = regexp.MustCompile(`\${([\p{L}\d\s\\\-_|.'":]*)}`)
```

Comment:
Small comment here, this mean the following variable declaration would be valid:
```
${dynami:c.lookup|'fallback.here'}
```
`dynami:c.lookup` seems weird to me, but again... this is still a valid key reference in YAML if you quote the key value.
```yaml
'test:ok': hello
```

Reply:
yeah! yaml? 😕

Reply:
I had yaml with a passion, not sure it was better than xml ;)